eth/catalyst/api.go

```diff
@@ -332,9 +332,13 @@ func computePayloadId(headBlockHash common.Hash, params *beacon.PayloadAttribute
 	return out
 }
 
-// invalid returns a response "INVALID" with the latest valid hash set to the current head.
-func (api *ConsensusAPI) invalid(err error) beacon.PayloadStatusV1 {
+// invalid returns a response "INVALID" with the latest valid hash supplied by latest or to the current head
+// if no latestValid block was provided.
+func (api *ConsensusAPI) invalid(err error, latestValid *types.Block) beacon.PayloadStatusV1 {
 	currentHash := api.eth.BlockChain().CurrentHeader().Hash()
+	if latestValid != nil {
```

Comment:
Btw, any particular reason for using CurrentHeader opposed to CurrentBlock? During snap sync, the former will keep moving forward vs the latter will only jump when sync completes.

Reply:
No, we should rather use current block since we want to return the latest validated block